hazelcast-client/src/test/java/com/hazelcast/client/ClientDisconnectTest.java

```diff
@@ -168,32 +168,6 @@ public void run() throws Exception {
         }, 3);
     }
 
-    @Test
-    public void testPendingInvocationAndWaitEntryCancelled_whenDisconnected_withLock() {
```

Comment:
Instead of deleting this kind of tests, I think we should replace them with IMap.lock.

Reply:
I was expecting that there's already a test with `IMap.lock` 😞

Reply:
We have more tests similar to this. Since both locks work the same, I was doing the tests with only one of them.

Reply:
https://github.com/hazelcast/hazelcast/pull/14901/files#diff-3fb94f96e140075cec3aadff2cc8a5e4
Hmm. It seems that we have tests for both of them also. Please be careful. These tests are valuable to us.

Reply:
You should be more careful, since you are the reviewer! 🤣

Reply:
Challenge accepted!  🤣